parcels/particlefile/baseparticlefile.py

```diff
@@ -92,12 +94,16 @@ def __init__(self, name, particleset, outputdt=np.infty, write_ondelete=False, c
             self.time_origin = self.particleset.time_origin
             self.lonlatdepth_dtype = self.particleset.collection.lonlatdepth_dtype
             self.var_names = []
+            self.var_dtypes = []
             self.var_names_once = []
+            self.var_dtypes_once = []
             for v in self.particleset.collection.ptype.variables:
-                if v.to_write == 'once':
+                if v.to_write == 'once' and self.write_ondelete is False:
```

Comment:
I don't understand why this change is needed. Does it not work without this change?

Reply:
No, it seems to crash when a particle is set to write `'once'` and `write_ondelete=True`. I got the same crash from the master branch. I'm not entirely sure what the cause of the crash is (and I do not know whether this is a niche crash with my complex configuration or something more basic) but this seemed to be an easy fix that worked.

Reply:
Hmm, but the change doesn't make sense from a conceptual angle: your changes to the `if/else` change the meaning of this statement

I suggest we return to the original code in this PR, and if needed you open another Issue on the bug you're experiencing